overlord/snapstate/snapstate.go

```diff
@@ -119,11 +119,23 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 	linkSnap := s.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q available to the system%s"), ss.Name, revisionStr))
 	addTask(linkSnap)
 
-	// do GC!
-	if snapst.HasCurrent() {
+	// Do not do that if we are reverting to a local revision
+	if ss.Revision.Unset() && snapst.HasCurrent() {
```

Comment:
I think this check is no longer going to be right in the new world (Revision is sometimes going to be set for installs as well, need to be !revisionIsLocal ? )